### PR TITLE
Add dashboard page

### PR DESCRIPTION
**Pull request by patrickjfl**
# Context

[Trello ticket ](https://trello.com/c/mm4AIPpI/674-product-homepage)

# Changes in this PR

This PR creates a dashboard page for all users, with cards rendered based on the user's role/s. These roles are extracted from the JWT in `res.locals.user.token`.

### Before

Screenshot 2023-12-19 at 15 51 16

### After

POM and admin view:

Screenshot 2023-12-19 at 15 47 50

Assessor view:

Screenshot 2023-12-19 at 15 48 16

# Release checklist

As part of our continuous deployment strategy we must ensure that this work is
ready to be released at any point. Before merging to `main` we must first
confirm:

## Pre merge checklist

- [ ] Are any changes required to the e2e tests?
- [ ] If you've added a new route, have you added a new
  `auditEvent`? (see `server/routes/apply.ts` for examples)
- [ ] Are there environment variables or other infrastructure configuration which needs to be included in this release?
- [ ] Are there any data migrations required. Automatic or manual?
- [ ] Does this rely on changes being deployed to the CAS API?

## Post merge checklist

Once we've merged it will be auto-deployed to the dev environment.

[Find the build-and-deploy job in CircleCI](https://app.circleci.com/pipelines/github/ministryofjustice/hmpps-community-accommodation-tier-2-ui).

Should a release fail at any step, you as the author should now lead the work to
fix it as soon as possible.

### CODE REVIEW

File: server/controllers/dashboardController.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import type { Request, RequestHandler, Response } from 'express'
+import { jwtDecode } from 'jwt-decode'
+import { sectionsForUser } from '../utils/userUtils'
+
+export default class DashboardController {
+  index(): RequestHandler {
+    return (_req: Request, res: Response) => {
+      const { authorities: roles = [] } = jwtDecode(res.locals.user.token) as { authorities?: string[] }
```
Comment (Harriethw) [inline server/controllers/dashboardController.ts:8]: ah this is cool, thought we would have to make another API call!

Comment (Harriethw) [reply server/controllers/dashboardController.ts:8]: Quite minor but the test coverage report is saying line 8 isn't covered by tests...

**Review by Harriethw — APPROVED** [review]
This is so cool! Great to see this finally in

File: server/controllers/dashboardController.ts (hunk 1 above)
Comment (patrickjfl) [reply server/controllers/dashboardController.ts:8]: credit to @edavey for spotting that we could use what was in the token!

Comment (patrickjfl) [reply server/controllers/dashboardController.ts:8]: is it important/valuable to mock jwtDecode to get that line covered, do you think?

Comment (edavey) [reply server/controllers/dashboardController.ts:8]: yes, I would say that it's useful documentation: for the unit tests to describe how the roles are extracted from `res.local.user.token` by `jwtDecode()`

Comment (patrickjfl) [reply server/controllers/dashboardController.ts:8]: ok, I have added a new test - let me know if you're happy with it!

**Pull request merged**